### PR TITLE
fix: fixed oddFunction to eval negative numbers

### LINKED ISSUE #151: odd() function doesn't work for negative numbers
**Describe the bug**
The `odd()` function produces a wrong result for negative numbers.

**To Reproduce**
Expression: `odd(-1)` returns `false` but should be `true`.

**Expected behavior**
`odd(-1)` returns `true`

**Environment**
* FEEL engine version: 1.11.2

### PR DESCRIPTION
**Pull request by kaaquist**
Signed-off-by: Kasper Aaquist Johansen <kasperaaquist@gmail.com>

## Description

* Made oddFunction return `true` when evaluating negative numbers.  

closes #151

### CODE REVIEW

File: src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala [hunk 1]
```diff
@@ -100,7 +101,7 @@ object NumericBuiltinFunctions {
 
   private def oddFunction =
     builtinFunction(params = List("number"), invoke = {
-      case List(ValNumber(n)) => ValBoolean(n % 2 == 1)
+      case List(ValNumber(n)) => ValBoolean(abs(n) % 2 == 1)
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala:104]: We can use the function `abs` directly on the `BigDecimal`. 

```suggestion
      case List(ValNumber(n)) => ValBoolean(n.abs % 2 == 1)
```

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala [hunk 2]
```diff
@@ -205,6 +205,8 @@ class BuiltinNumberFunctionsTest
 
     eval(" odd(5) ") should be(ValBoolean(true))
     eval(" odd(2) ") should be(ValBoolean(false))
+    eval(" odd(-5)") should be(ValBoolean(true))
+    eval(" odd(-2)") should be(ValBoolean(false))
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:209]: In order to separate the cases, we can create a new test function below. For example:

``` 
it should "return true if a negative number is odd" in {
    eval(" odd(-5)") should be(ValBoolean(true))
    eval(" odd(-2)") should be(ValBoolean(false))
}
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@kaaquist thank you for your contribution :cookie: 

Looks good :+1: Just two comments for improvements :)

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala (hunk 2 above)
Comment (kaaquist) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:213]: Separated code into its own test.

File: src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala (hunk 1 above)
Comment (kaaquist) [reply src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala]: Done. I did actually see that you also had a similar usages of `.abs` in the absFunction, so make sense.

File: src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala [hunk 3]
```diff
@@ -5,6 +5,7 @@ import org.camunda.feel.impl.builtin.BuiltinFunction.builtinFunction
 import org.camunda.feel.syntaxtree._
 
 import scala.math.BigDecimal.RoundingMode
+import scala.math.Numeric.BigDecimalAsIfIntegral.abs
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/builtin/NumericBuiltinFunctions.scala:8]: We can remove the unused import :)

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala [hunk 4]
```diff
@@ -207,6 +207,12 @@ class BuiltinNumberFunctionsTest
     eval(" odd(2) ") should be(ValBoolean(false))
   }
 
+  "A odd() function" should "return true if negative number is odd" in {
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:210]: Here, we can use the `it` function to combine both cases under one root :)

```suggestion
  it should "return true if negative number is odd" in {
```

**Review by saig0 — CHANGES_REQUESTED** [review]
Two tiny points left :candy:

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala (hunk 4 above)
Comment (kaaquist) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala]: done.

**Review by saig0 — APPROVED** [review]
Thanks @kaaquist :tada:

**Pull request merged**